server/src/main/java/org/opensearch/cluster/node/DiscoveryNodeFilters.java

```diff
@@ -69,18 +69,30 @@
         }
     };
 
-    public static DiscoveryNodeFilters buildFromKeyValue(OpType opType, Map<String, String> filters) {
-        Map<String, String[]> bFilters = new HashMap<>();
+    /**
+     * Creates or updates filters returning a new {@link DiscoveryNodeFilters} object.
+     * If the new object has no filters, {@code null} is returned.
+     */
+    @Nullable
+    public static DiscoveryNodeFilters buildOrUpdateFromKeyValue(
+        final DiscoveryNodeFilters original,
+        final OpType opType,
```

Comment:
The thing I don't like about this method signature is that `opType` is unused if `original` is non-null. What if this is called where `original.opType != opType`? Seems like opportunity for confusion. Unfortunately this seems like a consequence of using null to represent no filters and I don't really have a suggestion that avoids some code duplication. I would at least add a unit test that asserts the behavior if there is an opType mismatch.

Reply:
> The thing I don't like about this method signature is that opType is unused if original is non-null. What if this is called where original.opType != opType? Seems like opportunity for confusion. Unfortunately this seems like a consequence of using null to represent no filters and I don't really have a suggestion that avoids some code duplication.

Good catch! I also found it weird, but could not think of clean solution

> I would at least add a unit test that asserts the behavior if there is an opType mismatch.

I have added change to assert the behavior. Let me know what you think.